src/btree/bt_curnext.c

```diff
@@ -280,9 +284,22 @@ __cursor_var_next(WT_CURSOR_BTREE *cbt, bool newpage, bool restart)
                 --cbt->recno;
                 continue;
             }
-            WT_RET(__wt_page_cell_data_ref(session, page, &unpack, cbt->tmp));
 
+            cbt->slot = WT_COL_SLOT(page, cip);
```

Comment:
Could you explain what is changed here? Was `__wt_page_cell_data_ref` doing something that wasn't OK before?

Reply:
Previously __wt_page_cell_data_ref returns the onpage value assuming the onpage value is stable and should be visible at this point. However, this is wrong in durable history and we need to do a txn read instead as we always write the latest version to disk and that may not be visible.